ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -585,6 +594,17 @@ public func first<T, E>(producer: SignalProducer<T, E>) -> Result<T, E>? {
 	return result
 }
 
+/// Starts the producer, then blocks, waiting for the last value.
+public func last<T, E>(producer: SignalProducer<T, E>) -> Result<T, E>? {
+	return producer |> takeLast(1) |> single
+}
+
+/// Starts the producer, then blocks, waiting for completion.
+public func wait<T, E>(producer: SignalProducer<T, E>) -> Result<(), E> {
+	let result = producer |> map { _ in () } |> last
```

Comment:
> That would also avoid the need for `??` below, I think.

Don't think so, `concat`ing a static value doesn't make the type system statically know there is a value, so the `last` operator still returns an optional.